### ISSUE TITLE
add ASCII Liblouis Braille table

### ISSUE THREAD

**Issue by ainekc** [nvaccess/nvda#13312]
### Is your feature request related to a problem? Please describe.
Yeah, can't read BRF files properly without this if they are written in upper case (with dot 7s on the bottom) which a lot are. this impacts e.g. anyone wanting to read Braille music on a PC via opening a BRF in notepad or similar.
### Describe the solution you'd like
Add the ASCII Liblouis Braille table.
### Describe alternatives you've considered
I have been trying with 6 and 8 dot US computer Braille. With 8-dot, there's dot 7s on the letters that you can't get rid of. With 6-dot, there are 456, sometimes followed by 345, in front of letters, which I guess signify uppercase or such - they're attached to the letters. 

an alternative is to add a setting to display using 6 dots/ignoring dots 7 and 8, regardless of the Braille table selected - this way you could still use the 8-dot computer Braille table and it would display correctly. But seemingly the ASCII Braille table was designed for the correct encoding without fiddling with any of this so, yeah, I think that's the best option - I just tried it on IOS and it worked a charm.

**Comment by CyrilleB79** [comment]
Cc @aaclause

**Comment by feerrenrut** [comment]
If this is supported in the version of liblouis we are already using, we will look to expose it from NVDA. This has been added to our queue to confirm, and we'll implement if we can.

**Comment by cary-rowen** [comment]
Sorry, I want to know what's going on with this issue? Does liblouis support this feature?
@feerrenrut

**Comment by jrbowden** [comment]
Hi, this issue has recently come up again.

I think there are two main ways  to solve this:

1. NVDA can have a new option, like other screen readers, six/eight dot braille mode.
This is relatively easy to implement:
a) add a checkbox for this feature to Braille Settings
b) Add an appropriate gesture to quickly toggle the option (e.g. Braille Space+ 2 + 3 + 5)
c) When on (six dot), activate the following simple code:
- Take the output string after translation and mask off dots 7 and 8
- (typically, for each character: character becomes character & 0x3f).
- This is done before applying any highlighting or cursor, so cursor still uses dots 7 and 8.

2. Alternatively, it should be relatively easy to create the required Liblouis table - it does not currently exist.

My preferred option is to add the new option into NVDA - other screen readers have this feature and it would not be limited to one language.

Trust this helps.

**Comment by zstanecic** [comment]
Hi,

This has it’s downsides.

Some languages are using dots 8 to display numbers. 

How user should differentiate between this when having this option turned on?

**Comment by jrbowden** [comment]
The reason someone would want this option is deliberately to read braille in six-dot code, such as a BRF file. It is not useful for reading eight-dot computer code, for example as you say, when the table uses dot 8 for numbers. Dot 8 for numbers does not occur in six-dot BRF files.
Hope this makes sense.

**Comment by LeonarddeR** [comment]
I'm not entirely sure I understand the problem. Is BRF only intended for 6-dot braille? I thought it could also just contain 8-dot Braille. In that case, the US 8 dot table should provide for that, right?
If BRF is written in upper case which adds dots 7 to every letter, isn't that just an issue in the source file that needs to be fixed?
I know that handy tech devices have an in hardware feature to disable dots 7 and 8, so I guess it makes sense to add this as an option.

**Comment by Adriani90** [comment]
Yes I think adding an option to disable or enable dots 7 and 8 makes sense in my view, it is less depending on the language and most braille display vendors implement this in their drivers, or at least they can do it quite easily.

**Comment by seanbudd** [comment]
Apparently this table doesn't exist in Liblouis yet? A Liblouis issue needs to be opened for this to be progressable

**Comment by jrbowden** [comment]
Forgive me for butting in:
There are several ways that BRF tables are encoded, including US, German (Euro), French, Spanish, etc, etc.
Therefore, I propose it would be better to have an option in NVDA to mask off the bottom two dots. Other screen readers, e.g. JAWS, SuperNova, have a 6/8 dot checkbox, that can also be a keyboard or braille display command.

It is very simple:
If the user toggles 6 dot to on, then simply go through the braille string and mask (AND) with 0x3f, i.e. turn off dots 7 and 8.

Of course, this is done before adding the cursor.

I propose it is not a good idea to ask Liblouis to make a (single) six-dot table, because there are several that will have to be made.

As simple checkbox to control the feature in my opinion is a better solution.

Trust this helps.
